src/content/app/tools/blast/components/species-selector/SpeciesSelector.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { updateSelectedSpecies } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+import { getSelectedSpecies } from 'src/content/app/tools/blast/state/blast-form/blastFormSelectors';
+
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+import styles from './SpeciesSelector.scss';
+
+type SpeciesList = {
+  assembly_name: string;
+  common_name: string | null;
+  genome_id: string;
+  scientific_name: string;
+};
+
+export type Props = { speciesList: SpeciesList[] };
+
+const SpeciesSelector = (props: Props) => {
+  const { speciesList } = props;
+
+  speciesList.sort((a, b) => {
+    if (a.common_name) {
+      if (a.common_name && b.common_name) {
+        return a.common_name > b.common_name ? 1 : -1;
+      }
+      return -1;
+    } else {
+      if (!a.common_name && !b.common_name) {
+        return a.scientific_name > b.scientific_name ? 1 : -1;
+      }
+      return 1;
+    }
+  });
```

Comment:
1) `.sort` mutates the original array, and we **never** mutate props passed to a component
2) if we are using a hard-coded list of species, we don't even need to do the sorting; we can hard-code the order of species into the hard-coded list

Reply:
Good point about the props...do we know if we will need the sorting later on. it took me quite a bit to get the sorting right and will like to keep it if we are going to use it again. Just want to make sure we wont use it again. Also i have been testing the sorting functionality in my test script.

Reply:
Just seen your comment below, will go with the suggestion, sounds good.